File: hw/collahuazo/u3/hw24CompositePaterns/CompositePatternPy/composite/model/supervisor.py
from composite.model.employee import Employee

class Supervisor(Employee):

    def __init__(self):
        super().__init__()
        self.direct_reports = []

    def state_name(self):
        super().state_name()  
        if len(self.direct_reports) > 0:  
            for employee in self.direct_reports:
                employee.state_name()

    def add(self, an_employee):
        self.direct_reports.append(an_employee)

File: hw/aimacana/u3/hw25CompositePattern/java/Composite/src/main/java/ec/edu/espe/composite/model/Supervisor.java
package ec.edu.espe.composite.model;
import java.util.Vector;

public abstract class Supervisor extends Employee {
    protected Vector<Employee> directReports = new Vector<>();
    
    @Override
    public String stateName() {
        StringBuilder sb = new StringBuilder();
        sb.append(super.stateName()); 
        if (directReports.size() > 0) { 
            for (int i = 0; i < directReports.size(); ++i) {
                sb.append(((Employee)directReports.elementAt(i)).stateName());
            }
        }
        return sb.toString();
    }
    
    public void add(Employee anEmployee) {
        this.directReports.addElement(anEmployee);
    }
}
